### ISSUE TITLE
embedding widget in Sphinx HTML Documentation error

### ISSUE THREAD

**Issue by cornhundred** [jupyter-widgets/ipywidgets#1245]
I tried embedding a widget using either jupyter_sphinx or nbsphinx and am getting the same error each time on the console: 

```
Uncaught (in promise) Error: Cannot find module 'clustergrammer_widget'
```

in ``theme.js``. Screenshot below:
![screen shot 2017-03-30 at 6 08 36 pm](https://cloud.githubusercontent.com/assets/8352840/24529648/d855d754-157a-11e7-8f3b-2c9465499f6d.png)

I'm not getting any errors when I build with Sphinx and it appears to be able to load my widget properly. 

I also added the python libraries to the list of enabled extensions and to the requirements.txt for readthedocs hosting. I do not see this error on the console from the hosted version of the docs [link to docs](http://clustergrammer.readthedocs.io/clustergrammer_widget.html). Here is a link to the [rst file](https://github.com/MaayanLab/clustergrammer-docs/blob/master/docs/clustergrammer_widget.rst) on GitHub where I'm using the directive 

```
.. ipywidgets-display::

  # import classes and instantiate Network instance with the widget as an argument
  from clustergrammer_widget import *
  net = Network(clustergrammer_widget)

  # load matrix file
  net.load_file('rc_two_cats.txt')

  # cluster using default parameters
  net.cluster()

  # make interactive widget
  net.widget()

```

**Comment by SylvainCorlay** [comment]
FYI the embedder always looks for the module locallg before trying the CDN so having at least one error message per widget module loaded is expected.

You can check out that this occurs o  the jupyter.org/widgets page.

**Comment by cornhundred** [comment]
The error in ``utils.js`` line 74 (see screenshot above) occurs where ``utils.js`` tries to run 

``window.require([module_name], success_callback, failure_callback);`` 

and cannot find my custom module. 

```
if (requirejsDefined) {
    if (module_name !== 'jupyter-js-widgets' || window.requirejs.defined('jupyter-js-widgets')) {
        modulePromise = new Promise(function (innerResolve, innerReject) {
            var success_callback = function (module) {
                innerResolve(module);
            };
            var failure_callback = require_error ? require_error(success_callback, reject, module_version) : reject;
            window.require([module_name], success_callback, failure_callback);
        });
    }
    else if (module_name === 'jupyter-js-widgets') {
        modulePromise = Promise.resolve(require('../'));
    }
}

```

I am able to get the ``jupyter-js-widgets`` module to load (e.g. documentation [example](http://ipywidgets.readthedocs.io/en/latest/embedding.html#using-the-jupyter-sphinx-extension)).  In the case of loading ``jupyter-js-widgets`` it bypassses line 74 because it recognizes the name; see line 64: ``// If the module is jupyter-js-widgets, we can just self import.``.  

I'm not sure if my module is supposed to be loaded from unpkg.com, but I do not see it in the sources. The widget works in nbviewer and I can see it loading in the sources from unpkg.com. 

So, maybe the problem is that Sphinx is not loading my widget from unpkg.com. Or is there a way I can embed my widget JavaScript into the page if the CDN is not working?

**Comment by jasongrout** [comment]
By the way, I *think* the code that sphinx should be using to try to load the widget is here: https://github.com/jupyter-widgets/ipywidgets/blob/master/jupyter-js-widgets/src-embed/embed-manager.ts#L54

Basically, it calls hat error handler, which should try to load it from unpkg. Can you see if that code is available and running?

**Comment by cornhundred** [comment]
I'm not sure if this is what you are asking for @jasongrout, but this is what I see running from my sources: 

![screen shot 2017-04-11 at 5 23 30 pm](https://cloud.githubusercontent.com/assets/8352840/24931592/da0c0f5e-1edb-11e7-9879-76a17b200a91.png)

which looks similar to what you linked to above 

![screen shot 2017-04-11 at 5 26 18 pm](https://cloud.githubusercontent.com/assets/8352840/24931660/0def1668-1edc-11e7-96e0-2c49a1741454.png)

It looks like my compiled ``manager-base.js`` script is not trying to load from unpkg.

**Comment by willingc** [comment]
Closing this issue since it appears to be resolved in 7.6.3. Thanks to all!
